### ISSUE TITLE
can I improve accuracy?

### ISSUE THREAD

**Issue by nicosmaris** [jonasoreland/runnerup#971]
I seam to have worse accuracy than e.g. endomondo, can I do something about it? A setting in the app or in the phone? I have a samung A40.

I am walking in a building and it shows that I am walking mainly in the next building 😄 

If outdoor activities are accurate then ok but it says poor GPS and if I don't open [GPSTest](https://play.google.com/store/apps/details?id=com.android.gpstest) first, it keeping waiting fir GPS signal forever.

https://runalyze.com/athlete/nickmaris

**Comment by gerhardol** [comment]
Starting GPStest prior to an app may force update AGPS data, which helps the device to find satellites quicker using the mobile towers to find the location. (This is not implemented in any running tracker I know of.) If you have a recent fix, this is not needed. (I never do this.)

The most important is to have a good lock on many GPS satellites before starting. Do not move and stay in an open area, wait for Good GPS. You can get more details by expanding the notification, if the sats and accuracy numbers stabilize, you are better off.
This is also regardless of the running app (but the data shown may be limited, you may not know if you should wait).
This affects the accuracy.

It is limited what a device can do with the raw data, it cannot guess a better position, just smooth the data from zick-zack. You may not cross over the street so often, but consistently run through the buildings on one side of the street. Another benefit would be that the pace is more stable.

If the GPS is not very good, I would recommend that the sampling rate is limited to 2s or so, to let the device smooth some.
You can also smooth the data when saving/uploading the data with the simplification filter in RU.

I tried to compare tracks recorded with different apps some years ago and did not see any apps with magic smoothing algorithms (so Endomondo was the same as the other I tested with). This is where running watches shines, their algorithm is tuned to determine current pace rather than current position to give better current pace info as well as smooth the track. 

Current GPS can give an accuracy of 3-5m (reported by the manufacturer firmware, not to be trusted...). There are a few phones with dual GPS with L5/E5 signals that can improve the accuracy down to less than a meter. There are few real world investigations from what I have seen, but many phones with the hardware do not use the new information so the accuracy is not better. This should improve in some years though with new phones and new software.

**Comment by nicosmaris** [comment]
Indeed, not moving and being in an open area is a better starting point.

What is the app waiting to say that the signal is good? How many sats are
enough and which accuracy numbers shall I look?

How can I read and change the sampling rate?

What is RU?

**Comment by nicosmaris** [comment]
Also, is there any interest in developing a watch app? It could be limited
to tracking and feeding data to the phone.

**Comment by gerhardol** [comment]
RU: RunnerUp

8 or more sats and better than 7m accuracy (the accuracy should maybe be adjusted down a little).
https://github.com/jonasoreland/runnerup/blob/master/app/src/main/org/runnerup/view/StartActivity.java#L808
Other apps may have other limits, there is no standard.

Settings-Recording-top item to st sampling rate

Watch app: To develop a watch app you need a watch to develop on which I do not have.
Similar for the Wear companion in general, I assume it can be improved a lot.

In general, the apps deserves a maintainer that can put in more time. I spend most time on overhead like updating to Google's new requirements, investigating crashes etc not adding features.
(I primarily use the app to maintain it right now...)

**Comment by nicosmaris** [comment]
Thanks a lot for your quick and detailed feedback!

From my point of view this ticket can now be closed.
